--- libiberty/rust-demangle.c.origin	2021-08-07 11:46:57.462708723 +0000
+++ libiberty/rust-demangle.c	2021-08-07 11:47:36.757095745 +0000
@@ -75,10 +75,10 @@
   int version;
 
   /* Recursion depth.  */
-  uint recursion;
+  uint32_t recursion;
   /* Maximum number of times demangle_path may be called recursively.  */
 #define RUST_MAX_RECURSION_COUNT  1024
-#define RUST_NO_RECURSION_LIMIT   ((uint) -1)
+#define RUST_NO_RECURSION_LIMIT   ((uint32_t) -1)
 
   uint64_t bound_lifetime_depth;
 };
