diff -ur fltk-1.1.10/FL/Fl_Menu_Item.H bd_fltk/FL/Fl_Menu_Item.H
--- fltk-1.1.10/FL/Fl_Menu_Item.H	2005-04-16 08:13:17.000000000 +0800
+++ bd_fltk/FL/Fl_Menu_Item.H	2010-06-12 21:19:09.000000000 +0800
@@ -88,7 +88,7 @@
   void callback(Fl_Callback1*c, long p=0) {callback_=(Fl_Callback*)c; user_data_=(void*)p;}
   void* user_data() const {return user_data_;}
   void user_data(void* v) {user_data_ = v;}
-  long argument() const {return (long)user_data_;}
+  long argument() const {return (long)(intptr_t)user_data_;}
   void argument(long v) {user_data_ = (void*)v;}
   int shortcut() const {return shortcut_;}
   void shortcut(int s) {shortcut_ = s;}
diff -ur fltk-1.1.10/FL/Fl_Widget.H bd_fltk/FL/Fl_Widget.H
--- fltk-1.1.10/FL/Fl_Widget.H	2007-11-20 00:21:48.000000000 +0800
+++ bd_fltk/FL/Fl_Widget.H	2010-06-12 21:18:26.000000000 +0800
@@ -30,6 +30,8 @@
 
 #include "Enumerations.H"
 
+#include <stdint.h>
+
 class Fl_Widget;
 class Fl_Window;
 class Fl_Group;
@@ -155,7 +157,7 @@
   void callback(Fl_Callback1*c, long p=0) {callback_=(Fl_Callback*)c; user_data_=(void*)p;}
   void* user_data() const {return user_data_;}
   void user_data(void* v) {user_data_ = v;}
-  long argument() const {return (long)user_data_;}
+  long argument() const {return (long)(intptr_t)user_data_;}
   void argument(long v) {user_data_ = (void*)v;}
   Fl_When when() const {return (Fl_When)when_;}
   void when(uchar i) {when_ = i;}
diff -ur fltk-1.1.10/fluid/Fl_Widget_Type.cxx bd_fltk/fluid/Fl_Widget_Type.cxx
--- fltk-1.1.10/fluid/Fl_Widget_Type.cxx	2009-06-29 06:09:25.000000000 +0800
+++ bd_fltk/fluid/Fl_Widget_Type.cxx	2010-06-12 21:38:42.000000000 +0800
@@ -1133,7 +1133,7 @@
 {0}};
 
 void align_cb(Fl_Button* i, void *v) {
-  int b = int(long(i->user_data()));
+  int b = int((intptr_t)(i->user_data()));
   if (v == LOAD) {
     if (current_widget->is_menu_item()) {i->deactivate(); return;} else i->activate();
     i->value(current_widget->o->align() & b);
@@ -1244,7 +1244,7 @@
 // "v_attributes" let user type in random code for attribute settings:
 
 void v_input_cb(Fl_Input* i, void* v) {
-  int n = int(long(i->user_data()));
+  int n = int((intptr_t)(i->user_data()));
   if (v == LOAD) {
     i->static_value(current_widget->extra_code(n));
   } else {
diff -ur fltk-1.1.10/src/fl_open_uri.cxx bd_fltk/src/fl_open_uri.cxx
--- fltk-1.1.10/src/fl_open_uri.cxx	2007-01-24 04:45:28.000000000 +0800
+++ bd_fltk/src/fl_open_uri.cxx	2010-06-12 21:21:34.000000000 +0800
@@ -114,7 +114,7 @@
 #ifdef WIN32
   if (msg) snprintf(msg, msglen, "open %s", uri);
 
-  return (int)ShellExecute(HWND_DESKTOP, "open", uri, NULL, NULL, SW_SHOW) > 32;
+  return (int)(intptr_t)ShellExecute(HWND_DESKTOP, "open", uri, NULL, NULL, SW_SHOW) > 32;
 
 #elif defined(__APPLE__)
   char	*argv[3];			// Command-line arguments
diff -ur fltk-1.1.10/test/arc.cxx bd_fltk/test/arc.cxx
--- fltk-1.1.10/test/arc.cxx	2006-10-11 11:12:15.000000000 +0800
+++ bd_fltk/test/arc.cxx	2010-06-12 21:40:08.000000000 +0800
@@ -66,7 +66,7 @@
 
 void slider_cb(Fl_Widget* o, void* v) {
   Fl_Slider* s = (Fl_Slider*)o;
-  args[long(v)] = s->value();
+  args[(intptr_t)(v)] = s->value();
   d->redraw();
 }
 
diff -ur fltk-1.1.10/test/cursor.cxx bd_fltk/test/cursor.cxx
--- fltk-1.1.10/test/cursor.cxx	2006-10-11 11:12:15.000000000 +0800
+++ bd_fltk/test/cursor.cxx	2010-06-12 21:40:35.000000000 +0800
@@ -39,7 +39,7 @@
 Fl_Hor_Value_Slider *cursor_slider;
 
 void choice_cb(Fl_Widget *, void *v) {
-  cursor = (Fl_Cursor)(long)v;
+  cursor = (Fl_Cursor)(intptr_t)v;
   cursor_slider->value(cursor);
   fl_cursor(cursor,fg,bg);
 }
diff -ur fltk-1.1.10/test/curve.cxx bd_fltk/test/curve.cxx
--- fltk-1.1.10/test/curve.cxx	2006-10-11 11:12:15.000000000 +0800
+++ bd_fltk/test/curve.cxx	2010-06-12 21:40:50.000000000 +0800
@@ -83,7 +83,7 @@
 
 void slider_cb(Fl_Widget* o, void* v) {
   Fl_Slider* s = (Fl_Slider*)o;
-  args[long(v)] = s->value();
+  args[(intptr_t)(v)] = s->value();
   d->redraw();
 }
 
diff -ur fltk-1.1.10/test/unittests.cxx bd_fltk/test/unittests.cxx
--- fltk-1.1.10/test/unittests.cxx	2007-05-20 08:01:06.000000000 +0800
+++ bd_fltk/test/unittests.cxx	2010-06-12 21:39:42.000000000 +0800
@@ -37,7 +37,7 @@
 int point_test_ix, line_test_ix, rect_test_ix, viewport_test_ix, circle_test_ix;
 
 void changePageCB(Fl_Widget*, void *ixvp) {
-  long ix = (long)ixvp;
+  long ix = (intptr_t)ixvp;
   long i = 0, n = win->children();
   for ( ; i<n; i++) 
     win->child(i)->hide();
