--- Makefile.origin	2020-02-09 03:31:52.814314289 +0100
+++ Makefile	2020-02-09 03:38:06.588112742 +0100
@@ -28,12 +28,12 @@
 LIBPOINT=0
 LIBVER=$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
 
-SOURCES = dgif_lib.c egif_lib.c gifalloc.c gif_err.c gif_font.c \
+SOURCES = dgif_lib.c egif_lib.c gifalloc.c gif_font.c \
 	gif_hash.c openbsd-reallocarray.c
 HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
 OBJECTS = $(SOURCES:.c=.o)
 
-USOURCES = qprintf.c quantize.c getarg.c 
+USOURCES = qprintf.c quantize.c getarg.c gif_err.c
 UHEADERS = getarg.h
 UOBJECTS = $(USOURCES:.c=.o)
 
@@ -92,7 +92,8 @@
 install: all install-bin install-include install-lib install-man
 install-bin: $(INSTALLABLE)
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
-	$(INSTALL) $^ "$(DESTDIR)$(BINDIR)"
+	# $(INSTALL) $^ "$(DESTDIR)$(BINDIR)"
+	$(INSTALL) $(foreach bin, $(INSTALLABLE), $(bin).exe) "$(DESTDIR)$(BINDIR)"
 install-include:
 	$(INSTALL) -d "$(DESTDIR)$(INCDIR)"
 	$(INSTALL) -m 644 gif_lib.h "$(DESTDIR)$(INCDIR)"
