--- lib/ExecutionEngine/JIT/JIT.cpp.origin	2013-08-03 10:20:36.000000000 +0800
+++ lib/ExecutionEngine/JIT/JIT.cpp	2013-08-03 10:21:58.000000000 +0800
@@ -69,7 +69,7 @@
 
 // Determine whether we can register EH tables.
 #if (defined(__GNUC__) && !defined(__ARM_EABI__) && \
-     !defined(__USING_SJLJ_EXCEPTIONS__))
+     !(defined(__USING_SJLJ_EXCEPTIONS__) || defined(WIN64)))
 #define HAVE_EHTABLE_SUPPORT 1
 #else
 #define HAVE_EHTABLE_SUPPORT 0
--- lib/ExecutionEngine/MCJIT/SectionMemoryManager.cpp.origin	2013-08-03 10:25:46.000000000 +0800
+++ lib/ExecutionEngine/MCJIT/SectionMemoryManager.cpp	2013-08-03 10:26:12.000000000 +0800
@@ -148,7 +148,7 @@
 
 // Determine whether we can register EH tables.
 #if (defined(__GNUC__) && !defined(__ARM_EABI__) && \
-     !defined(__USING_SJLJ_EXCEPTIONS__))
+     !(defined(__USING_SJLJ_EXCEPTIONS__) || defined(WIN64)))
 #define HAVE_EHTABLE_SUPPORT 1
 #else
 #define HAVE_EHTABLE_SUPPORT 0
--- tools/clang/lib/Frontend/InitHeaderSearch.cpp.origin	2013-08-04 14:32:27.000000000 +0800
+++ tools/clang/lib/Frontend/InitHeaderSearch.cpp	2013-08-04 14:34:32.000000000 +0800
@@ -399,6 +399,11 @@
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.6.2");
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.6.3");
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.0");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.1");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.2");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.3");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.9.0");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.9.1");
     // mingw.org C++ include paths
     AddMinGWCPlusPlusIncludePaths("/mingw/lib/gcc", "mingw32", "4.5.2"); //MSYS
 #if defined(_WIN32)
