--- pthread.h.origin	2010-02-24 17:16:01.000000000 +0800
+++ pthread.h	2010-02-24 17:41:35.000000000 +0800
@@ -302,10 +302,15 @@
 #endif
 #endif
 
+/* mingw-w64 may include <windows.h> which define this */
+#ifdef _TIMESPEC_DEFINED
+#define HAVE_STRUCT_TIMESPEC 1
+#endif
+
 #ifndef HAVE_STRUCT_TIMESPEC
 #define HAVE_STRUCT_TIMESPEC 1
 struct timespec {
-        long tv_sec;
+        time_t tv_sec;
         long tv_nsec;
 };
 #endif /* HAVE_STRUCT_TIMESPEC */
