--- src/pugiconfig.hpp.origin	2013-03-10 09:34:35.000000000 +0800
+++ src/pugiconfig.hpp	2013-03-10 09:35:20.000000000 +0800
@@ -21,10 +21,10 @@
 // #define PUGIXML_NO_XPATH
 
 // Uncomment this to disable STL
-// #define PUGIXML_NO_STL
+#define PUGIXML_NO_STL
 
 // Uncomment this to disable exceptions
-// #define PUGIXML_NO_EXCEPTIONS
+#define PUGIXML_NO_EXCEPTIONS
 
 // Set this to control attributes for public classes/functions, i.e.:
 // #define PUGIXML_API __declspec(dllexport) // to export all public symbols from DLL
