--- qtdeclarative/src/qmldevtools/qmldevtools.pro.orign	2016-08-21 11:46:36.136683039 +0800
+++ qtdeclarative/src/qmldevtools/qmldevtools.pro	2016-08-21 11:46:49.261119260 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 TARGET     = QtQmlDevTools
 QT         = core-private
 CONFIG += static internal_module qmldevtools_build
--- qtdeclarative/tools/qmlmin/qmlmin.pro.origin	2016-08-21 13:26:06.522379379 +0800
+++ qtdeclarative/tools/qmlmin/qmlmin.pro	2016-08-21 13:26:13.830633371 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 QT       = core qmldevtools-private
 SOURCES += main.cpp
 
--- qtdeclarative/tools/qmlimportscanner/qmlimportscanner.pro.origin	2016-08-21 13:28:44.127855039 +0800
+++ qtdeclarative/tools/qmlimportscanner/qmlimportscanner.pro	2016-08-21 13:28:49.664047315 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 
 QT = core qmldevtools-private
 DEFINES += QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_ASCII
--- qtdeclarative/tools/qmllint/qmllint.pro.origin	2016-08-21 13:33:30.753973586 +0800
+++ qtdeclarative/tools/qmllint/qmllint.pro	2016-08-21 13:33:42.018384999 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 
 QT = core qmldevtools-private
 
--- qtdeclarative/tools/qmlcachegen/qmlcachegen.pro.origin      2018-01-26 22:44:44.892133490 +0800
+++ qtdeclarative/tools/qmlcachegen/qmlcachegen.pro     2018-01-26 22:45:58.278763373 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 
 QT = qmldevtools-private
 DEFINES += QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_ASCII
--- qttools/src/linguist/lupdate/lupdate.pro.origin	2016-08-24 16:43:36.075198042 +0800
+++ qttools/src/linguist/lupdate/lupdate.pro	2016-08-24 16:43:44.611497063 +0800
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 QT = core-private
 
 qtHaveModule(qmldevtools-private) {
--- qttools/src/qdoc/qdoc.pro.origin	2016-08-24 22:13:16.694454533 +0800
+++ qttools/src/qdoc/qdoc.pro	2016-08-24 22:13:27.690845751 +0800
@@ -3,7 +3,7 @@
     requires(!contains(QT_DISABLED_FEATURES, xmlstreamwriter))
 }
 
-option(host_build)
+# option(host_build)
 QT = core
 qtHaveModule(qmldevtools-private) {
     QT += qmldevtools-private
--- qtdeclarative/tools/qmlformat/qmlformat.pro.origin	2020-09-14 14:42:13.106776253 +0000
+++ qtdeclarative/tools/qmlformat/qmlformat.pro	2020-09-14 14:45:37.990637907 +0000
@@ -1,4 +1,4 @@
-option(host_build)
+# option(host_build)
 
 QT = core qmldevtools-private
 
