--- src/msw/textentry.cpp.origin	2013-03-29 14:36:50.000000000 +0100
+++ src/msw/textentry.cpp	2013-03-29 14:36:45.000000000 +0100
@@ -60,7 +60,7 @@
 
 #if defined(__MINGW32__) || defined (__WATCOMC__) || defined(__CYGWIN__)
     // needed for IID_IAutoComplete, IID_IAutoComplete2 and ACO_AUTOSUGGEST
-    #include <shlguid.h>
+//    #include <shlguid.h>
 
     #ifndef ACO_AUTOAPPEND
         #define ACO_AUTOAPPEND 0x02
