--- configure.orig	2011-08-28 21:22:40.000000000 +0800
+++ configure	2011-08-28 21:22:58.000000000 +0800
@@ -241,7 +241,7 @@
       CFLAGS="$CFLAGS -arch x86_64"
       LDFLAGS="$LDFLAGS -arch x86_64"
     elif [ "$SYS" = MINGW ]; then
-      ASFLAGS="$ASFLAGS -f win32 -m amd64 -DPREFIX"
+      ASFLAGS="$ASFLAGS -f win32 -m amd64"
     else
       ASFLAGS="$ASFLAGS -f elf -m amd64"
     fi
--- Makefile.orig	2011-08-28 21:59:30.000000000 +0800
+++ Makefile	2011-08-28 21:59:46.000000000 +0800
@@ -17,7 +17,6 @@
 ifneq ($(AS),)
 X86SRC = common/i386/cpu-a.asm \
          common/i386/dct-sse2.asm \
-         common/i386/deblock.asm \
          common/i386/deblock_inter.asm \
          common/i386/deblock_intra.asm \
          common/i386/mc-a01.asm \
--- common/cpu.c.orig	2011-08-28 22:21:22.000000000 +0800
+++ common/cpu.c	2011-08-28 22:21:04.000000000 +0800
@@ -39,6 +39,7 @@
 #include <machine/cpu.h>
 #endif
 
+#include "config.h"
 #include "common.h"
 #include "cpu.h"
 
@@ -199,6 +200,12 @@
   return cpu;
 }
 
+void
+xavs_cpu_restore (uint32_t cpu)
+{
+	  return;
+}
+
 #elif defined( ARCH_PPC )
 
 #if defined(SYS_MACOSX) || defined(SYS_OPENBSD)
