--- src/lib_json/json_reader.cpp.origin	2016-08-22 08:32:02.410672854 +0800
+++ src/lib_json/json_reader.cpp	2016-08-22 08:32:23.100177494 +0800
@@ -44,7 +44,7 @@
 namespace Json {
 
 #if __GNUC__ >= 6
-typedef std::scoped_ptr<CharReader> const  CharReaderPtr;
+typedef std::unique_ptr<CharReader> const  CharReaderPtr;
 #else
 typedef std::auto_ptr<CharReader>          CharReaderPtr;
 #endif
--- src/lib_json/json_writer.cpp.origin	2016-08-22 08:33:54.916416961 +0800
+++ src/lib_json/json_writer.cpp	2016-08-22 08:34:09.290767699 +0800
@@ -55,7 +55,7 @@
 namespace Json {
 
 #if __GNUC__ >= 6
-typedef std::scoped_ptr<StreamWriter> const  StreamWriterPtr;
+typedef std::unique_ptr<StreamWriter> const  StreamWriterPtr;
 #else
 typedef std::auto_ptr<StreamWriter>          StreamWriterPtr;
 #endif
